python_modules/dagster/dagster/_core/definitions/reconstruct.py

```diff
@@ -260,7 +260,9 @@ def subset_for_execution(
     ) -> "ReconstructablePipeline":
         # take a list of unresolved selection queries
         check.opt_list_param(solid_selection, "solid_selection", of_type=str)
-        check.opt_set_param(asset_selection, "asset_selection", of_type=AssetKey)
+        asset_selection = frozenset(
+            check.opt_set_param(asset_selection, "asset_selection", of_type=AssetKey)
+        )
```

Comment:
this is a behavior change - before `None` would flow through now we coerce in to empty set. Pretty sure None vs empty set is meaningful so dont think we want this change.